implementation/map-to-scatter-gather/README.md

```diff
@@ -0,0 +1,118 @@
+
+# Replace Map with Scatter-Gather
+
+This project is the CDK implementation of ['Replace Map with Scatter-Gather'](https://serverlessland.com/refactoring-serverless/map-to-scatter-gather) pattern. It shows how yo can use SNS in CDK to replace AWS Step Functions `Map` state to implement the [scatter-gather](https://www.enterpriseintegrationpatterns.com/patterns/messaging/BroadcastAggregate.html) pattern.
+
+## How it works
+
+The code will deploy two stacks:
+
+1. ScatterGatherStack
+2. RefactoredScatterGather
+
+Both stacks deploy 3 lambda functions, ```requester```, ```responder``` and ```aggregator```. The ```requester``` sends a message(quote) with a generated ```uuid```, ```responder``` processes the message and outputs a modified message containing a randomly generated ```price_quote```. The ```aggregator``` aggregates the received quotes. Lambda application code is re-used in both stacks.
+
+The first stack deploys a step function workflow implementing the scatter-gather pattern using the map state. When triggered the workflow sends a quote request that is processed by 4 parallel ```responder``` in a map state. The ```responder``` adds price_quotes. When all parallel executions have completed the result is provided aggregated to the ```aggregator``` function.
```

Comment:
- I'd refer to the stack by name
- Then you can shorten the description to "..deploys a Step Functions [caps] workflow using a map state to send a quote request to 4 ```responder``` instances."